### PR TITLE
Clean up txHandler and transactionPool.

### PR DESCRIPTION
**Pull request by derbear**

### CODE REVIEW

File: data/pools/transactionPool.go [hunk 1]
```diff
@@ -295,15 +304,15 @@ func (pool *TransactionPool) Remember(txgroup []transactions.SignedTxn) error {
 	pool.mu.Lock()
 	defer pool.mu.Unlock()
 
+	if pool.pendingBlockEvaluator == nil {
+		return fmt.Errorf("TransactionPool.Remember: no pending block evaluator")
+	}
+
```
Comment (tsachiherman) [inline data/pools/transactionPool.go]: this wouldn't hurt, but it's a duplicate. it's the first thing being done in pool.test

File: data/txHandler.go [hunk 2]
```diff
@@ -166,38 +166,44 @@ func (handler *TxHandler) backlogWorker() {
...
-				handler.net.Disconnect(wi.rawmsg.Sender)
-				continue
-			}
-
-			// we've processed this message, so increase the counter.
-			transactionMessagesHandled.Inc(nil)
-
-			// at this point, we've verified the transaction, so we can safely treat the transaction as a verified transaction.
-			verifiedTxGroup := wi.unverifiedTxGroup
+			handler.postprocessCheckedTxn(wi)
```
Comment (tsachiherman) [inline data/txHandler.go]: I think that you meant to replace the upper switch statement with this call as well, right ? (postprocessCheckedTxn )

File: data/pools/transactionPool.go (hunk 1 above)
Comment (derbear) [reply data/pools/transactionPool.go]: I just moved a block of code a few lines. Maybe we should delete it.

File: data/txHandler.go (hunk 2 above)
Comment (derbear) [reply data/txHandler.go]: Ah, yeah, there was some git issue.

Comment (derbear) [reply data/txHandler.go]: FYI There was also a metric that wasn't being logged correctly before. This should fix that.

**Pull request merged**